module/Activity/src/Activity/Controller/AdminCategoryController.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+namespace Activity\Controller;
+
+use Zend\Mvc\Controller\AbstractActionController;
+use Zend\Session\Container as SessionContainer;
+use Zend\View\Model\ViewModel;
+
+class AdminCategoryController extends AbstractActionController
+{
+    /**
+     * View all Categories.
+     */
+    public function indexAction()
+    {
+        $categoryService = $this->getServiceLocator()->get('activity_service_category');
+        $categories = $categoryService->getAllCategories();
+
+        return ['categories' => $categories];
+    }
+
+    /**
+     * Add Category.
+     */
+    public function addAction()
+    {
+        $categoryService = $this->getServiceLocator()->get('activity_service_category');
+        $form = $categoryService->getCategoryForm();
+        $request = $this->getRequest();
+        $translator = $this->getServiceLocator()->get('translator');
+
+        if ($request->isPost()) {
+            $form->setData($request->getPost());
+
+            if ($form->isValid()) {
+                $category = $categoryService->createCategory($form->getData());
```

Comment:
Why not handle the form in the category service, as is customary in the rest of the code base?